guides/source/getting_started.md

```diff
@@ -7,1150 +7,1548 @@ This guide covers getting up and running with Ruby on Rails.
+  Product Update (0.2ms)  UPDATE "products" SET "name" = 'T-Shirt', "updated_at" = '2024-11-09 22:39:09.693548' WHERE "products"."id" = 1 /*application='Store'*/
+  TRANSACTION (0.0ms)  COMMIT TRANSACTION /*application='Store'*/
+=> true
 ```
 
-And when we want to fetch all articles from the database, we can call [`all`](
-https://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-all)
-on the model:
+### Deleting Records
+
+The `destroy` method can be used to delete a record from the database.
 
 ```irb
-blog(dev)> Article.all
-=> #<ActiveRecord::Relation [#<Article id: 1, title: "Hello Rails", body: "I am on Rails!", created_at: "2020-01-18 23:47:30", updated_at: "2020-01-18 23:47:30">]>
+store(dev)> product.destroy
+  TRANSACTION (0.1ms)  BEGIN immediate TRANSACTION /*application='Store'*/
+  Product Destroy (0.4ms)  DELETE FROM "products" WHERE "products"."id" = 1 /*application='Store'*/
+  TRANSACTION (0.1ms)  COMMIT TRANSACTION /*application='Store'*/
+=> #<Product:0x0000000125813d48 id: 1, name: "T-Shirt", created_at: "2024-11-09 22:39:38.498730000 +0000", updated_at: "2024-11-09 22:39:38.498730000 +0000">
 ```
 
-This method returns an [`ActiveRecord::Relation`](
-https://api.rubyonrails.org/classes/ActiveRecord/Relation.html) object, which
-you can think of as a super-powered array.
+This deleted the T-Shirt product from our database. We can confirm this with `Product.all` to see that that it only returns Pants.
 
-TIP: To learn more about models, see [Active Record Basics](
-active_record_basics.html) and [Active Record Query Interface](
-active_record_querying.html).
+```irb
+store(dev)> Product.all
+  Product Load (1.9ms)  SELECT "products".* FROM "products" /* loading for pp */ LIMIT 11 /*application='Store'*/
+=>
+[#<Product:0x000000012abde4c8
+  id: 2,
+  name: "Pants",
+  created_at: "2024-11-09 22:33:19.638912000 +0000",
+  updated_at: "2024-11-09 22:33:19.638912000 +0000">]
+```
 
-Models are the final piece of the MVC puzzle. Next, we will connect all of the
-pieces together.
+### Validations
 
-### Showing a List of Articles
+Active Record provides *validations* which allows you to ensure data inserted into the database adheres to certain rules.
 
-Let's go back to our controller in `app/controllers/articles_controller.rb`, and
-change the `index` action to fetch all articles from the database:
+Let's add a `presence` validation to the Product model to ensure that all products must have a `name`.
 
 ```ruby
-class ArticlesController < ApplicationController
-  def index
-    @articles = Article.all
-  end
+class Product < ApplicationRecord
+  validates :name, presence: true
 end
 ```
 
-Controller instance variables can be accessed by the view. That means we can
-reference `@articles` in `app/views/articles/index.html.erb`. Let's open that
-file, and replace its contents with:
+Let's try to create a Product without a name in the Rails console.
```

Comment:
You may have to use the `reload!` command to refresh the console before testing the validation.